### PR TITLE
As per the feedback, updated code for more reusability, efficiency

### PR DESCRIPTION
**Pull request by ChandrashekarSukuri**

### CODE REVIEW

File: e2e/page-objects/add-users-pageobj.ts [hunk 1]
```diff
@@ -23,43 +25,61 @@ export class AddUsers{
...
+  public addUsersHeading = element(by.id('addUsersHeading'));
+
   async getAddUserForm(){
     await browser.get('http://localhost:4200/');
   }
 
-  getheadingText(){
-    return element(by.id('addUsersHeading')).getText();
+  getHeadingText(){
+    return this.addUsersHeading.getText();
```
Comment (narora2) [inline e2e/page-objects/add-users-pageobj.ts:35]: 1. Wait for text to be present before returning the text
2. Should be async

File: e2e/page-objects/add-users-pageobj.ts [hunk 2]
```diff
@@ -23,43 +25,61 @@ export class AddUsers{
...
   async clickGenderOptionByName(name:string){
-    await element(by.css('[type="radio"][name="gender"][value="'+name+'"]')).click();
+    const ele = element(by.css('[type="radio"][name="gender"][value="'+name+'"]'));
+    await this.waitUtil.waitForElementToBePresent(ele);
+    return ele.click();
   }
 
   getDesignationOptionsByName(name:string){
-    return  element(by.cssContainingText('[role="listbox"] mat-option.designation-option',name));
+    return element(by.cssContainingText('[role="listbox"] mat-option.designation-option',name));
```
Comment (narora2) [inline e2e/page-objects/add-users-pageobj.ts:51]: Same as above

File: e2e/page-objects/add-users-pageobj.ts [hunk 3]
```diff
@@ -23,43 +25,61 @@ export class AddUsers{
...
   }
 
   getDesignationFilterOptionsByName(name:string){
     return element(by.cssContainingText('[role="listbox"] mat-option.designation-filter-option',name));
```
Comment (narora2) [inline e2e/page-objects/add-users-pageobj.ts:55]: Same as above

File: e2e/page-objects/magento-pageObject.ts [hunk 4]
```diff
@@ -133,14 +148,19 @@ export class MagentoPageObject {
   }
 
   async getNewProductsNames(){
-    return await this.productsNamesLinks.map(async (ele)=> await ele.getText());
+    return this.productsNamesLinks.map(async (ele)=> await ele.getText());
```
Comment (narora2) [inline e2e/page-objects/magento-pageObject.ts:151]: Same as above. You can wait for the 1st element in the array to be present

File: e2e/page-objects/magento-pageObject.ts [hunk 5]
```diff
@@ -133,14 +148,19 @@ export class MagentoPageObject {
...
   }
 
   async getNewProductPrices(){
-    return await this.productPrices.map(ele => ele.getText());
+    return this.productPrices.map(ele => ele.getText());
```
Comment (narora2) [inline e2e/page-objects/magento-pageObject.ts:155]: Same as above

File: e2e/page-objects/magento-pageObject.ts [hunk 6]
```diff
@@ -133,14 +148,19 @@ export class MagentoPageObject {
...
   }
 
   async getOrderIds(){
-    return await this.ordersTableIdsColumns.map(ele => ele.getText());
+    return this.ordersTableIdsColumns.map(ele => ele.getText());
```
Comment (narora2) [inline e2e/page-objects/magento-pageObject.ts:159]: Same as above

**Comment by ChandrashekarSukuri** [comment]
I will make the changes mentioned Neha

________________________________

**Comment by narora2** [comment]
Thanks Chandra

From: ChandrashekarSukuri ***@***.***>
Date: Tuesday, August 22, 2023 at 7:11 AM
To: ChandrashekarSukuri/protractor-learning ***@***.***>
Cc: Neha Arora ***@***.***>, Mention ***@***.***>
Subject: Re: [ChandrashekarSukuri/protractor-learning] As per the feedback, updated code for more reusability, efficiency (PR #11)
I will make the changes mentioned Neha

________________________________

**Pull request merged**